### DISCUSSION TITLE
MAUI context menu support

### DISCUSSION THREAD

**Discussion by vadimffe** [HavenDV/H.NotifyIcon#95] [Q&A]
Hello, I see there is an example of NET maui tray icon implementation, but is there an example of an implementation of tray icon with a context menu?

**Comment by HavenDV** [comment]
Maui is still not supported. But you can use H.NotifyIcon core library in Windows in Maui, example in H.NotifyIcon.Apps.Console project
https://github.com/HavenDV/H.NotifyIcon/tree/master/src/apps/H.NotifyIcon.Apps.Console

**Comment by HavenDV** [comment]
I released a MAUI package and fixed some bugs and now it should work for MAUI for simple cases. Context menu and bindings are supported. Sample app here: https://github.com/HavenDV/H.NotifyIcon/tree/master/src/apps/H.NotifyIcon.Apps.Maui
I want to note that the TrayIcon icon should be located in the Resources / Raw directory, and not in Resources / Images

**Comment by vadimffe** [comment]
I just had time to test it and it seems to be working fine. Question I still have left, do you have some idea how to make NET MAUI app to minimize into system tray? I mean on clicking minimize application would go into system tray without icon staying on a task bar?

**Reply by HavenDV** [reply]
I added extensions for Window type - ShowInTaskbar and HideInTaskbar
They will be available in 10-15 minutes on NuGet
You can use them to achieve the desired behavior.

**Reply by vadimffe** [reply]
Everything seems to work pretty fine. However I am not able to figure out how to override default Minimize window logic to hide application the same time. Your library has functionality to `window.Hide();`, but I am having hard times to get this action also trigger once I am clicking minimize button on my application window. Do you have some hints?

**Reply by HavenDV** [reply]
Unfortunately no. Typically for a window close button this is done by intercepting the Closing event and setting Cancel = true or overriding the OnClosing method. And perhaps there is a similar way to minimize
